### PR TITLE
Remove unused folder_id field from search table

### PR DESCRIPTION
**Pull request by demiankatz**
The search table in the database has contained a folder_id field since the early days of VuFind 1.x; however, this field has never been used for anything. This PR removes the unused field and associated code/indexes.

TODO
- [x] Add upgrade script logic to optionally remove field/index from MySQL (allow it to be kept in case somebody has found a local use)
- [x] Add migration to remove field/index from PostgreSQL
- [x] Run full test suite
- [x] Update changelog
- [x] Make appropriate updates to #2233 after merging.

### CODE REVIEW

**Comment by demiankatz** [comment]
@EreMaijala, I think this is more or less ready to go. The screen for selecting whether or not to remove the deprecated column in the upgrade process could probably be beautified, but at least it's functional. Maybe @crhallberg has some ideas there.

**Review by EreMaijala — APPROVED** [review]
Looks good to me and works as well!

**Comment by demiankatz** [comment]
Thanks, @EreMaijala! Here's a screen shot of the new upgrade screen in case @crhallberg wants to suggest any changes to confirmdeprecatedcolumns.phtml before I merge this PR:

image

File: themes/bootstrap3/templates/upgrade/confirmdeprecatedcolumns.phtml
Comment (crhallberg) [inline themes/bootstrap3/templates/upgrade/confirmdeprecatedcolumns.phtml]: As regular link, it looks like it will take you to documentation on how to do this. I would instead style these links as buttons by adding the classes `btn btn-default`.

**Review by crhallberg — CHANGES_REQUESTED** [review]

File: themes/bootstrap3/templates/upgrade/confirmdeprecatedcolumns.phtml
Comment (demiankatz) [reply themes/bootstrap3/templates/upgrade/confirmdeprecatedcolumns.phtml]: Thanks, @crhallberg. I had to use `btn btn-default btn-link` to avoid having underlines inside the buttons. Is that appropriate, or should I do things differently? (If I leave off `btn-default` they don't look like buttons).

Comment (crhallberg) [reply themes/bootstrap3/templates/upgrade/confirmdeprecatedcolumns.phtml]: That makes sense, I'm surprised that `.btn-default` doesn't remove the underline. That might be a more consistent option.

Comment (crhallberg) [reply themes/bootstrap3/templates/upgrade/confirmdeprecatedcolumns.phtml]: Same applies to #2741.

Comment (demiankatz) [reply themes/bootstrap3/templates/upgrade/confirmdeprecatedcolumns.phtml]: Please take a look at my latest changes to #2741. If you approve of that approach, I can update this to match once it is merged.

Comment (demiankatz) [reply themes/bootstrap3/templates/upgrade/confirmdeprecatedcolumns.phtml]: Okay, @crhallberg, #2741 has been merged into this, and I've adjusted the template accordingly. I think this is ready to merge now; just needs your approval.

**Review by crhallberg — APPROVED** [review]

**Pull request merged**
